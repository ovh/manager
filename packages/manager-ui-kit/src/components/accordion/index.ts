export { Accordion } from './Accordion.component';
export type { AccordionProps } from './Accordion.props';

export { AccordionContent } from './accordion-content/AccordionContent.component';
export { AccordionItem } from './accordion-item/AccordionItem.component';
export { AccordionTrigger } from './accordion-trigger/AccordionTrigger.component';
