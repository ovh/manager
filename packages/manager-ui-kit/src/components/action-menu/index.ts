export { ActionMenu } from './ActionMenu.component';
export type { ActionMenuItem, ActionMenuProps } from './ActionMenu.props';
