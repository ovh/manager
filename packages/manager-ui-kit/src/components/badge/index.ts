export { Badge } from './badge.component';

export type { BadgeProps } from './badge.props';

export * from './Badge.constants';
