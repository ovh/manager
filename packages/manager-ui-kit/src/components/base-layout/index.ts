export { BaseLayout } from './BaseLayout.component';

export type { BaseLayoutProps } from './BaseLayout.props';

export type { HeaderProps } from './header/Header.props';
