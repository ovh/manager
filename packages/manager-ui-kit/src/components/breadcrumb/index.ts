export { Breadcrumb } from './Breadcrumb.component';
export type { BreadcrumbProps } from './Breadcrumb.props';
export * from './Breadcrumb.constants';
