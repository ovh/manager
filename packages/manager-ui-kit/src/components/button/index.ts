export { Button } from './Button.component';
export type { ButtonProps } from './Button.props';
