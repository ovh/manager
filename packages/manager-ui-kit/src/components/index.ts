export * from './action-banner';
export * from './action-menu';
export * from './redirection-guard';
export * from './breadcrumb';
export * from './base-layout';
export * from './clipboard';
export * from './step';
export * from './tabs';
export * from './tiles-input-group';
export * from './tiles-input';

export * from './Link';
export * from './drawer';
export * from './guide-menu';

export * from './notifications';

export * from './price';

export * from './filters';

export * from './button';
export * from './text';

export * from './service-state-badge';

export * from './tile';
export * from './order';

export * from './badge';
export * from './modal';
export * from './tags-list';
export * from './tags-tile';
export * from './changelog-menu';
export * from './link-card';
export * from './error';
export * from './error-boundary';
export * from './update-name-modal';
export * from './delete-modal';
export * from './grid-layout';
export * from './onboarding-layout';
export * from './datagrid';
export * from './accordion';
export * from './checkbox';
export * from './combobox';
export * from './datepicker';
export * from './file-upload';
export * from './form-field';
export * from './message';
export * from './phone-number';
export * from './popover';
export * from './progress';
export * from './quantity';
export * from './radio-group';
export * from './select';
export * from './switch';
export * from './timepicker';
export * from './toggle';
export * from './tooltip';
export * from './tree-view';
