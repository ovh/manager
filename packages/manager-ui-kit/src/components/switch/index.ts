export { Switch } from './Switch.component';
export type { SwitchProps } from './Switch.props';

export { SwitchItem } from './switch-item/SwitchItem.component';
export * from './constants/SwitchConstants';
