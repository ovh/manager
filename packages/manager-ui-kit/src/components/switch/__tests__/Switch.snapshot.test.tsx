import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Switch, SwitchItem } from '@/components';

describe('Switch Snapshot tests', () => {
  it('renders the component with default props', () => {
    const { container } = render(
      <Switch>
        <SwitchItem value="item-1">Item 1</SwitchItem>
        <SwitchItem value="item-2">Item 2</SwitchItem>
        <SwitchItem value="item-3">Item 3</SwitchItem>
      </Switch>,
    );
    expect(container).toMatchSnapshot();
  });
});
