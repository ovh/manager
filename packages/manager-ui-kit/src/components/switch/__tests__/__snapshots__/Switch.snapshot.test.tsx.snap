// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch Snapshot tests > renders the component with default props 1`] = `
<div>
  <div
    class="_switch_r3co3_2 _switch--md_r3co3_14"
    data-ods="switch"
    data-orientation="horizontal"
    data-part="root"
    data-scope="toggle-group"
    dir="ltr"
    id="toggle-group::r0:"
    role="radiogroup"
    style="outline: none;"
    tabindex="0"
  >
    <button
      aria-checked="false"
      class="_switch-item_15c64_2"
      data-ods="switch-item"
      data-orientation="horizontal"
      data-ownedby="toggle-group::r0:"
      data-part="item"
      data-scope="toggle-group"
      data-state="off"
      dir="ltr"
      id="toggle-group::r0::item-1"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Item 1
    </button>
    <button
      aria-checked="false"
      class="_switch-item_15c64_2"
      data-ods="switch-item"
      data-orientation="horizontal"
      data-ownedby="toggle-group::r0:"
      data-part="item"
      data-scope="toggle-group"
      data-state="off"
      dir="ltr"
      id="toggle-group::r0::item-2"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Item 2
    </button>
    <button
      aria-checked="false"
      class="_switch-item_15c64_2"
      data-ods="switch-item"
      data-orientation="horizontal"
      data-ownedby="toggle-group::r0:"
      data-part="item"
      data-scope="toggle-group"
      data-state="off"
      dir="ltr"
      id="toggle-group::r0::item-3"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Item 3
    </button>
  </div>
</div>
`;
